docs/artifact_repositories/npm.md

```diff
@@ -58,5 +58,66 @@ npm error notarget In most cases you or one of your dependencies are requesting
 npm error notarget a package version that doesn't exist.
 ```
 
+If you're using `pnpm`, the output will look like this:
+
+```text
+ ERR_PNPM_NO_VERSIONS  No versions available for malicious. The package may be unpublished.
+
+This error happened while installing a direct dependency of /tmp/testing
+```
+
 If a version range is accepted by the manifest, the package manager will
 automatically attempt to use a version that passes Phylum's policy.
+
+### `yarn`
+
+Custom NPM registries can be configured with `yarn`:
+
+```sh
+yarn config set -H npmRegistryServer "https://npm.phylum.io/"
+yarn config set -H npmAuthIdent "<PHYLUM_ORG>/<PHYLUM_GROUP:<PHYLUM_API_KEY>"
+yarn config set -H npmAlwaysAuth true
+```
+
+> ⚠️ **WARNING** ⚠️
+>
+> Do not accidentally save your token into your shell history.
+
+A blocked package will show up in `yarn` output as missing:
+
+```text
+➤ YN0027: malicious@unknown can't be resolved to a satisfying range
+➤ YN0001: TypeError: Cannot read properties of undefined (reading 'dist')
+    at Fv.getCandidates (/home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:688:7154)
+    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
+    at async Pg.getCandidates (/home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:141:1271)
+    at async uH (/home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:401:9441)
+    at async /home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:401:8776
+    at async C (/home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:401:7127)
+    at async T2 (/home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:401:8456)
+    at async /home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:402:531
+    at async Promise.all (index 0)
+    at async /home/chris/.cache/node/corepack/v1/yarn/4.5.3/yarn.js:402:488
+
+➤ Errors happened when preparing the environment required to run this command.
+
+TODO: OR MAYBE THIS?
+
+➤ YN0027: malicious@unknown can't be resolved to a satisfying range
+➤ YN0035: Package not found
+➤ YN0035:   Response Code: 404 (Not Found)
+➤ YN0035:   Request Method: GET
+➤ YN0035:   Request URL: https://npm.phylum.io/malicious
+
+➤ Errors happened when preparing the environment required to run this command.
+```
```

Comment:
Currently still have this PR as draft because I'm not entirely convinced what to do about this.

The top error is what Aviary gives right now, which doesn't seem ideal.

The bottom error is what happens if we return a 404 instead of returning the package metadata with no versions. It seems like `yarn` doesn't expect an entirely empty list, but NPM and PNPM both seem to handle it just fine.

So do we change things to make Yarn happy, or do we just accept this more verbose error?

Reply:
The top error seems like a bug to me... Maybe we should open an issue with yarn?

The bottom error is cleaner output, but I don't think it's any more helpful to the developer... So I think we should continue to return an empty list to get the benefits for npm and pnpm users.

Reply:
I've opened a bug with Yarn, so we can see if they agree with that assessment:
https://github.com/yarnpkg/berry/issues/6630